core/src/main/java/io/grpc/transport/AbstractStream.java

```diff
@@ -104,6 +123,23 @@ public void deliverFrame(WritableBuffer frame, boolean endOfStream, boolean flus
   @Nullable
   public abstract IdT id();
 
+  /**
+   * The number of queued bytes for a given stream, below which {@link StreamListener#onReady(int)}
+   * will be called. Defaults to {@link #DEFAULT_ONREADY_THRESHOLD}.
+   */
+  public int getOnReadyThreshold() {
+    return onReadyThreshold;
+  }
+
+  /**
+   * Sets the number of queued bytes for a given stream, below which {@link StreamListener#onReady(int)}
+   * will be called. If not called, defaults to {@link #DEFAULT_ONREADY_THRESHOLD}.
+   */
+  public void setOnReadyThreshold(int onReadyThreshold) {
+    checkArgument(onReadyThreshold > 0, "onReadyThreshold must be > 0");
+    this.onReadyThreshold = onReadyThreshold;
```

Comment:
This needs to check whether onReady() needs to be called, or it needs to have restrictions on when it can be called.